### PR TITLE
core: add signed and unsigned types for sync duties

### PR DESCRIPTION
**Pull request by dB2510**
Adds signed and unsigned types for sync committee duties.

category: feature
ticket: #1177

### CODE REVIEW

File: testutil/random.go [hunk 1]
```diff
@@ -522,6 +551,15 @@ func RandomBitList() bitfield.Bitlist {
 	return resp
 }
 
+func RandomBitVec() bitfield.Bitvector128 {
+	size := 128
+	index := rand.Intn(size)
+	resp := bitfield.NewBitvector128()
```
Comment (xenowits) [inline testutil/random.go:557]:
```suggestion
		var (
			size  = 128
			index = rand.Intn(size)
			resp  = bitfield.NewBitvector128()
		)
```

File: testutil/random.go [hunk 2]
```diff
@@ -375,6 +375,35 @@ func RandomAggregateAndProof() *eth2p0.AggregateAndProof {
...
+		Contribution:    RandomSyncCommitteeContribution(),
+		SelectionProof:  RandomEth2Signature(),
+	}
+}
+
+func RandomSyncCommitteeContribution() *altair.SyncCommitteeContribution {
+	return &altair.SyncCommitteeContribution{
+		Slot:              RandomSlot(),
+		BeaconBlockRoot:   RandomRoot(),
+		SubcommitteeIndex: 0,
```
Comment (xenowits) [inline testutil/random.go:401]: why not a random index?

File: testutil/random.go [hunk 3]
```diff
@@ -375,6 +375,35 @@ func RandomAggregateAndProof() *eth2p0.AggregateAndProof {
...
+	}
+}
+
+func RandomCoreSyncContribution() core.SyncContribution {
+	return core.SyncContribution{SyncCommitteeContribution: *RandomSyncCommitteeContribution()}
+}
+
+func RandomSyncContributionAndProof() *altair.ContributionAndProof {
+	return &altair.ContributionAndProof{
+		AggregatorIndex: 0,
```
Comment (xenowits) [inline testutil/random.go:391]: why not a random index? Like

`AggregatorIndex: rand.Uint64()`

**Review by xenowits — APPROVED** [review]

File: testutil/random.go (hunk 3 above)
Comment (dB2510) [reply testutil/random.go]: done

File: testutil/random.go (hunk 2 above)
Comment (dB2510) [reply testutil/random.go]: done

**Pull request merged**